src/mock_crypto.rs

```diff
@@ -0,0 +1,242 @@
+        let new_rng = XorShiftRng::from_rng(other)?;
+        let _ = RNG.with(move |rng| rng.replace(new_rng));
+        Ok(())
+    }
+
+    /// Mock cryptographic functions.
+    pub mod crypto {
+        /// Mock signing.
+        pub mod sign {
+            use super::super::with_rng;
+            use rand::Rng;
+            use std::ops::{Index, RangeFull};
+
+            /// Number of bytes in a `PublicKey`.
+            pub const PUBLICKEYBYTES: usize = 32;
+            /// Number of bytes in a `SecretKey`.
+            pub const SECRETKEYBYTES: usize = 32;
+            /// Number of bytes in a `Signature`.
+            pub const SIGNATUREBYTES: usize = 32;
+
+            /// Mock signing public key.
+            #[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd,
+                     Serialize)]
+            pub struct PublicKey(pub [u8; PUBLICKEYBYTES]);
+
+            impl Index<RangeFull> for PublicKey {
+                type Output = [u8];
+                fn index(&self, index: RangeFull) -> &[u8] {
+                    self.0.index(index)
+                }
+            }
+
+            /// Mock signing secret key.
+            #[derive(Clone, Debug, Eq, PartialEq)]
+            pub struct SecretKey(pub [u8; SECRETKEYBYTES]);
+
+            /// Mock signature.
+            #[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, Ord, Serialize, PartialEq,
+                     PartialOrd)]
+            pub struct Signature(pub [u8; SIGNATUREBYTES]);
+
+            impl AsRef<[u8]> for Signature {
+                fn as_ref(&self) -> &[u8] {
+                    &self.0
+                }
+            }
+
+            /// Generate mock public and corresponding secret key.
+            pub fn gen_keypair() -> (PublicKey, SecretKey) {
+                with_rng(|rng| {
+                    let value = rng.gen();
+                    (PublicKey(value), SecretKey(value))
+                })
+            }
+
+            /// Sign a message using the mock secret key.
+            pub fn sign_detached(m: &[u8], sk: &SecretKey) -> Signature {
+                let mut temp = m.to_vec();
+                temp.extend(&sk.0);
+                Signature(hash256(&temp))
+            }
+
+            /// Verify the mock signature against the message and the signer's mock public key.
+            pub fn verify_detached(signature: &Signature, m: &[u8], pk: &PublicKey) -> bool {
+                let mut temp = m.to_vec();
+                temp.extend(&pk.0);
+                *signature == Signature(hash256(&temp))
+            }
+
+            fn hash256(data: &[u8]) -> [u8; 32] {
+                use tiny_keccak::sha3_256;
+                sha3_256(data)
+            }
+        }
+
+        /// Mock encryption.
+        pub mod box_ {
+            use super::super::with_rng;
+            use rand::Rng;
+
+            /// Number of bytes in a `PublicKey`.
+            pub const PUBLICKEYBYTES: usize = 32;
```

Comment:
Do we need these to be this big ? Can 8 be not sufficient ? So everything in mock is `[u8; 8]` for e.g.